src/main/java/com/atlassian/bitbucket/jenkins/internal/client/BitbucketClientFactory.java

```diff
@@ -46,6 +41,14 @@
      */
     BitbucketProjectSearchClient getProjectSearchClient();
 
+    /**
+     * Return a repository search client
+     *
+     * @param projectKey the key of the project to perform the search over
+     * @return a client that is ready to use
+     */
+    BitbucketRepositoryClient getRepositoryClient(String projectKey);
```

Comment:
Happy to change it back, but it seemed strange to me that the project client interface had methods to get other clients as well as to get the project itself.
Maybe changing the method name from `get()` to `getProject()` would help?
Lets hear what Gaurav thinks

Reply:
This is a tough one. I like the fluent one more but with Kristy modification of renaming get to getProject. We will need to rename others as well like get in the repository to getRepository. I don't think there was any significant benefit to having `T get()`. The reason I like fluent one more is because it mimics the hierarchial model of bitbucket server.  Also. we can potentially improve the URL building in future. For example, webhook don't need to build a URL from the root, it could simple append to one from the repository. 
> seemed strange to me that the project client interface had methods to get other clients as well as to get the project itself.

This is one of the side effects of fluent interface.